diff --git a/node_modules/recoil/cjs/recoil.js b/node_modules/recoil/cjs/recoil.js
index 0429381..8d56d96 100644
--- a/node_modules/recoil/cjs/recoil.js
+++ b/node_modules/recoil/cjs/recoil.js
@@ -4,7 +4,6 @@ Object.defineProperty(exports, '__esModule', { value: true });
 
 function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }
 
-var gkx = _interopDefault(require('gkx'));
 var reactDom = _interopDefault(require('react-dom'));
 var react = require('react');
 var react__default = _interopDefault(react);
@@ -1115,6 +1114,7 @@ Recoil_gkx.setFail = gk => {
 };
 
 var Recoil_gkx_1 = Recoil_gkx; // @oss-only
+var gkx = Recoil_gkx;
 
 /**
  * Copyright (c) Facebook, Inc. and its affiliates.
diff --git a/node_modules/recoil/native/recoil.js b/node_modules/recoil/native/recoil.js
index 61fb1de..8699509 100644
--- a/node_modules/recoil/native/recoil.js
+++ b/node_modules/recoil/native/recoil.js
@@ -1,4 +1,3 @@
-import gkx from 'gkx';
 import reactNative from 'react-native';
 import react, { useRef as useRef$2 } from 'react';
 
@@ -1108,6 +1107,7 @@ Recoil_gkx.setFail = gk => {
 };
 
 var Recoil_gkx_1 = Recoil_gkx; // @oss-only
+var gkx = Recoil_gkx_1;
 
 /**
  * Copyright (c) Facebook, Inc. and its affiliates.
@@ -3094,7 +3094,7 @@ function shouldNotBeFrozen(value) {
   } // Some environments, just as Jest, don't work with the instanceof check
 
 
-  if (!isSSR && (value === window || value instanceof Window)) {
+  if (!isSSR && (value === window)) {
     return true;
   }
 
